syntax_tree_nodes/include/declarations.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+#include <string>
+
+#include "types.h"
+#include "syntax_node.h"
+
+
+class FuncDecl;
+class Statement;
+
+class Declaration : virtual public SyntaxNode, ChildSyntaxNode<0> {
+public:
+    std::string GetDeclName() const {
+        return name_;
+    }
+
+    decl_type_t GetDeclType() const {
+        return decl_type_;
+    }
+
+protected:
+    Declaration(decl_type_t decl_type, const std::string &name) : SyntaxNode(syntax_node_t::DECL),
+    ChildSyntaxNode<0>(), decl_type_(decl_type), name_(name) {}
+
+    decl_type_t decl_type_;
+    std::string name_;
+};
+
+class VarDecl : public Declaration {
+public:
+    VarDecl(const std::string &var_name, const VarType *var_type, size_t bit_length) :
+    Declaration(decl_type_t::VAR_DECL, var_name), var_type_(var_type), bit_length_(bit_length) {}
+
+    const VarType *GetVarType() const {
+        return var_type_;
+    }
+
+    size_t GetBitLength() const {
+        return bit_length_;
+    }
+
+protected:
+    const VarType *var_type_;
+    size_t bit_length_;
```

Comment:
I think bit length need to be part of the type

Reply:
Done